packages/http-client-csharp/generator/Microsoft.Generator.CSharp/test/TestHelpers/MockHelpers.cs

```diff
@@ -48,6 +50,18 @@ public static Mock<CodeModelPlugin> LoadMockPlugin(
             mockPlugin.SetupGet(p => p.TypeFactory).Returns(mockTypeFactory.Object);
 
             CodeModelPlugin.Instance = mockPlugin.Object;
+
+            if (customization == null)
+            {
+                var workspace = new AdhocWorkspace();
+                Project project = workspace.AddProject("ExistingCode", LanguageNames.CSharp);
+                customization = project.GetCompilationAsync().GetAwaiter().GetResult()!;
```

Comment:
It would be better to avoid sync over async.

Reply:
I wonder if this class could be async, let me try

Reply:
Oh this is a method used in so many places, if i change this to async, all of them will have to changed to async as well as those `Setup` methods. Would this be OK?

Reply:
well I removed this part so that this is no longer an issue.